fictadvisor-web/src/app/(main)/schedule/schedule-page/calendar-section/components/groups-dropdown/GroupsDropDown.tsx

```diff
@@ -14,9 +15,20 @@ export const GroupsDropDown: FC<DropDownSectionProps> = ({ groups }) => {
     groupId: state.groupId,
   }));
   const setGroupId = useSetGroupId();
+  const [inputGroupId, setInputGroupId] = useState('');
+  const [isEditing, setIsEditing] = useState(false);
+
+  useEffect(() => {
+    setInputGroupId(groups.find(({ id }) => id === groupId)?.code || '');
+  }, [groupId, setInputGroupId]);
+
+  function handleInput(text: string) {
```

Comment:
Instead of that you could just use the groupId from localStorage or whatever place it is saved as a initialValue, without need to add a props to COMMON ui component

Reply:
alternatively, you could implement a debounce function to handle input changes, allowing the user to enter a value and only saving groupId after a delay